apps/src/turtle/turtle.js

```diff
@@ -1059,19 +1061,54 @@ Artist.prototype.step = function(command, values, options) {
       break;
     case 'stamp':
       var img = this.stamps[values[0]];
-      var width = img.width / 2;
-      var height = img.height / 2;
-      var x = this.x - width / 2;
-      var y = this.y - height / 2;
-      if (img.width !== 0) {
-        this.ctxScratch.drawImage(img, x, y, width, height);
-      }
+
+      var dimensions = scaleToBoundingBox(MAX_STICKER_SIZE, img.width, img.height);
+      var width = dimensions.width;
+      var height = dimensions.height;
+
+      // Rotate the image such the the turtle is at the center of the bottom of
+      // the image and the image is pointing (from bottom to top) in the same
+      // direction as the turtle.
+      this.ctxScratch.save();
+      this.ctxScratch.translate(this.x, this.y);
+      this.ctxScratch.rotate(utils.degreesToRadians(this.heading));
+      this.ctxScratch.drawImage(img, -width / 2, -height, width, height);
+      this.ctxScratch.restore();
+
       break;
   }
 
   return tupleDone;
 };
 
+/**
+ * Given the width and height of a rectangle this scales the dimensions
+ * proportionally such that neither is larger than a given maximum size.
+ *
+ * @param maxSize - The maximum size of either dimension
+ * @param width - The current width of a rectangle
+ * @param height - The current height of a rectangle
+ * @return an object containing the scaled width and height.
+ */
+function scaleToBoundingBox(maxSize, width, height) {
+  if (width < maxSize && height < maxSize) {
+    return [width, height];
```

Comment:
Ah shit, nice catch! Was originally returning the array, but when doc'ing the function decided an object would be better / more explicit. All of the images are larger than the bounding box so that line never got exercised. Oops!